src/producer.rs

```diff
@@ -254,20 +257,19 @@ impl<P: Partitioner> Producer<P> {
               V: AsBytes
     {
         let mut rs = try!(self.send_all(ref_slice(rec)));
-        if self.config.required_acks == 0 {
+
+        if self.config.required_acks != 0 {
             // ~ with no required_acks we get no response and
             // consider the send-data request blindly as successful
-            Ok(())
-        } else {
             assert_eq!(1, rs.len());
-            rs.pop().unwrap().offset.map(|_| ())
+            rs.pop().unwrap();
         }
+
+        Ok(())
     }
 
     /// Synchronously send all of the specified messages to Kafka.
```

Comment:
Can you please add a _"To validate that all of the specified records have been successfully delivered, inspection of the offsets on the returned confirms is necessary."_ here?